provider/kubernetes/kubernetes.go

```diff
@@ -278,6 +305,35 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
 	return &templateObjects, nil
 }
 
+func loadAuthCredentials(namespace, secretName string, k8sClient Client) ([]string, error) {
+	secret, ok, err := k8sClient.GetSecret(namespace, secretName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch secret %q/%q: %s", namespace, secretName, err)
+	}
+	if !ok {
+		return nil, fmt.Errorf("secret %q/%q not found", namespace, secretName)
+	}
+	if secret == nil {
+		return nil, errors.New("secret data must not be nil")
+	}
+	if len(secret.Data) != 1 {
+		return nil, errors.New("secret must contain single element only")
+	}
+	var firstElement []byte
+	for _, v := range secret.Data {
+		firstElement = v
+		break
+	}
```

Comment:
Why not `firstElement := secret.Data[0]`? We have already asserted there's only a single element.